portal/assignment.py

```diff
@@ -292,3 +281,72 @@ def assignment_grades():
 
         return render_template('layouts/teacher/assignments/assignment-grades.html', assignment=assignment, assignment_name=assignment_name)
     return redirect(url_for('teacher.home'))
+@bp.route('grades/grade-book', methods=('GET', 'POST'))
+@login_required
+@admin
+def grade_view():
+    if request.method == 'POST':
+        session = request.form['gradebook']
+        with db.get_db() as con:
+            with con.cursor() as cur:
+                cur.execute("""
+                SELECT s.id, u.id AS user_id, u.first_name, u.last_name
+                FROM sessions s JOIN roster r
+                ON s.id = r.session_id
+                JOIN users u
+                ON r.student_id = u.id
+                WHERE s.id = %s
+                """, (session))
+                students = cur.fetchall()
+        return render_template('grade/gradebook.html', students=students)
```

Comment:
Make sure to move your html files to fit the new project structure and update the render template here and on line 351